### PR TITLE
Create a script for devs to run services in custom dc mode.

### PR DESCRIPTION
**Pull request by keyurva**
* The script by default connects to a dedicated dc dev cloud sql and gcs folder.
* A custom dc instance that points at these is also deployed in the cloud here: https://dc-dev-kqb7thiuka-uc.a.run.app/
  + We can use this instance for custom dc CI / CD soon.

### CODE REVIEW

File: .run_cdc_dev.env [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# Copyright 2023 Google LLC
```
Comment (shifucun) [inline .run_cdc_dev.env:1]: nit: 2024

File: .run_cdc_dev.env [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+# Set to false is not using Cloud SQL.
+USE_CLOUDSQL=true
+CLOUDSQL_INSTANCE=datcom-website-dev:us-central1:dc-dev
+DB_USER=root
+
+# GCS or local folder to "user data" folder.
+# This is primarily for NL embeddings.
+# The code looks for it under $USER_DATA_PATH/datacommons/nl/custom_embeddings.yaml
+USER_DATA_PATH=gs://customdc-data/dc-dev
+GOOGLE_CLOUD_PROJECT=datcom-website-dev
```
Comment (shifucun) [inline .run_cdc_dev.env:36]: nit: put GOOGLE_CLOUD_PROJECT above line 28

File: .run_cdc_dev.env [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+# If using sqlite, set USE_CLOUDSQL to false and USE_SQLITE to true.
+USE_SQLITE=false
+SQLITE_PATH=
+
+# Set this to false if you don't want to start the NL server.
+ENABLE_MODEL=true
+
+# Change this to https://autopush.api.datacommons.org if local mixer needs to remote into autopush mixer.
+# If you change this, change DC_API_KEY as well.
+DC_API_ROOT=https://api.datacommons.org
```
Comment (shifucun) [inline .run_cdc_dev.env:48]: maybe put this right after line 21 since it has to change with DC_API_KEY?

File: nl_server/__init__.py [hunk 4]
```diff
@@ -36,6 +37,11 @@ def create_app():
         "[%(asctime)s][%(levelname)-8s][%(filename)s:%(lineno)s] %(message)s ",
         datefmt="%H:%M:%S",
     )
+
+  log_level = logging.WARNING
+  if is_debug_mode():
+    log_level = logging.INFO
+  logging.getLogger('werkzeug').setLevel(log_level)
   logging.getLogger('werkzeug').setLevel(logging.WARNING)
```
Comment (shifucun) [inline nl_server/__init__.py:45]: remove line 45?

File: nl_server/loader.py [hunk 5]
```diff
@@ -174,7 +182,8 @@ def _maybe_load_custom_dc_yaml():
 # (deploy/nl/).
 #
 def get_env_path(flask_env: str, file_name: str) -> str:
-  if flask_env in ['local', 'test', 'integration_test', 'webdriver']:
+  if flask_env in ['local', 'test', 'integration_test', 'webdriver'
+                  ] or is_custom_dc_dev():
```
Comment (shifucun) [inline nl_server/loader.py:186]: why not making custom_dc_dev a flask env?

File: run_cdc_dev.sh [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+# Make sure the following are taken care of before running this script:
+# - run ./run_npm.sh
+# - run ./scripts/update_git_submodules.sh
+# - run ./run_test.sh --setup_python
+
+# Kill forked processes on exit.
+trap "trap - SIGTERM && kill -- -$$" SIGINT SIGTERM EXIT
+
+source .run_cdc_dev.env && export $(sed '/^#/d' .run_cdc_dev.env | cut -d= -f1)
```
Comment (shifucun) [inline run_cdc_dev.sh:29]: suggest to print out current website, mixer and import git hash with bright color

File: run_cdc_dev.sh [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+  echo "DC_API_KEY not specified."
+  exit 1
+fi
+
+if [[ $MAPS_API_KEY == "" ]]; then
+  echo "MAPS_API_KEY not specified."
+  exit 1
+fi
+
+echo "DC_API_KEY = $DC_API_KEY"
```
Comment (shifucun) [inline run_cdc_dev.sh:42]: maybe curl https://[autopush].api.datacommons.org/v2/node?key=<KEY>&nodes=geoId/06&property=<-

and make sure the key and endpoints match.

Comment (keyurva) [reply run_cdc_dev.sh:49]: Done.

File: .run_cdc_dev.env (hunk 2 above)
Comment (shifucun) [reply .run_cdc_dev.env:39]: does this have anything to do with CLOUDSQL? if it's only for GCS maybe say GCS_PROJECT?

**Review by shifucun — APPROVED** [review]

File: .run_cdc_dev.env (hunk 2 above)
Comment (keyurva) [reply .run_cdc_dev.env:39]: The gcloud sdk looks for this variable so not something we can change.

**Review by keyurva — COMMENTED** [review]
Thanks for the review.

**Pull request merged**